filechooser-portal/Main.vala

```diff
@@ -218,22 +217,20 @@ public class Files.FileChooserPortal : Object {
         var _results = new HashTable<string, Variant> (str_hash, str_equal);
         uint _response = 2;
 
-        dialog.destroy.connect (() => {
-            if (dialog.register_id != 0) {
-                connection.unregister_object (dialog.register_id);
-            }
-        });
-
         dialog.response.connect ((id) => {
             switch (id) {
                 case Gtk.ResponseType.OK:
-                    _results["uris"] = dialog.get_uris ();
-                    _results["choices"] = dialog.get_choices ();
-                    if (dialog.filter != null) {
-                        _results["current_filter"] = dialog.filter.to_gvariant ();
-                    }
+                    if (check_overwrite (dialog.get_file ()) == Gtk.ResponseType.OK) {
```

Comment:
Yes, it would be annoying for a warning to keep being shown when the application has already granted permission.  So the warning should only be shown when the uri has been chosen *within the filechooser*.  I'll pass the uri instead.